src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php

```diff
@@ -65,15 +65,34 @@ private function findAndSortTaggedServices(string|TaggedIteratorArgument $tagNam
             $class = $definition->getClass();
             $class = $container->getParameterBag()->resolveValue($class) ?: null;
             $reflector = null !== $class ? $container->getReflectionClass($class) : null;
-            $checkTaggedItem = !$definition->hasTag($definition->isAutoconfigured() ? 'container.ignore_attributes' : $tagName);
+            $loadFromDefaultMethods = $reflector && null !== $defaultPriorityMethod;
+
+            if ($reflector && $definition->isAutoconfigured() && !$definition->hasTag('container.ignore_attributes')) {
+                $loadFromAttributes = $attributes !== $tagAttributes = array_filter($attributes);
```

Comment:
As soon as one tag is listed with no tag-attributes, we load php-attributes.
php-attributes are always loaded first. This allows explicit tags to always win over php-attributes when indexes are explicitly declared.